src/classes/HouseholdNamingService.cls

```diff
@@ -0,0 +1,512 @@
+    * @return void
+    */
+    public void updateHouseholdNameAndMemberCount(List<Id> householdOrAccountIds){
+        if (isHouseholdNamingDisabled) {
+            return;
+        }
+        //we need this turned on to prevent recursive triggering on household creation
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.flag.HH, true);
+
+        List<SObject> householdsOrAccounts = householdsOrAccountsFor(householdOrAccountIds);
+        Map<Id, List<Contact>> membersByHouseholdId =
+                getHouseholdMembersByHouseholdId(householdOrAccountIds);
+        setHouseholdNameFieldValues(householdsOrAccounts, membersByHouseholdId);
+        setNumberOfHouseholdMembers(householdsOrAccounts, membersByHouseholdId);
+        save(householdsOrAccounts);
+
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.flag.HH, false);
+    }
+
+    private void save(List<SObject> householdsOrAccounts) {
+        if (!householdsOrAccounts.isEmpty()) {
+            UTIL_DMLService.updateRecords(householdsOrAccounts);
+        }
+    }
+
+    private void setNumberOfHouseholdMembers(List<SObject> householdsOrAccounts, Map<Id, List<Contact>> membersByHouseholdId) {
+        for (SObject household : householdsOrAccounts) {
+            List<Contact> householdMembers = membersByHouseholdId.get(idFor(household));
+            household.put(UTIL_Namespace.StrTokenNSPrefix('Number_of_Household_Members__c'),
+                    householdMembers != null ? householdMembers.size() : 0);
+        }
+    }
+
+    private void setHouseholdNameFieldValues(List<SObject> householdsOrAccounts,
+            Map<Id, List<Contact>> membersByHouseholdId) {
+        for (SObject household : householdsOrAccounts) {
+            List<Contact> householdMembers =
+                    membersByHouseholdId.get(idFor(household)) == null ?
+                            new List<Contact>() :
+                            membersByHouseholdId.get(idFor(household));
+            if (settings.isAdvancedHouseholdNaming()) {
+                setNameFieldValuesOnHousehold(
+                        household,
+                        new HouseholdName(
+                                new Contacts(householdMembers),
+                                namingService));
+            }
+        }
+    }
+
+    private Map<Id, List<Contact>> getHouseholdMembersByHouseholdId(List<Id> householdOrAccountIds) {
+        return householdMembersByHouseholdId(householdMembersFor(householdOrAccountIds));
+    }
+
+    private List<SObject> householdsOrAccountsFor(List<Id> householdOrAccountIds) {
+        List<SObject> hhupdatelist = new List<SObject>();
+        hhupdatelist.addAll(householdsFor(householdOrAccountIds));
+        hhupdatelist.addAll(accountsFor(householdOrAccountIds));
+        return hhupdatelist;
+    }
+
+    private List<Account> accounts;
+    private List<SObject> accountsFor(List<Id> hhids) {
```

Comment:
```suggestion
    private List<SObject> getAccountsFor(List<Id> hhids) {
```

Though, both of these methods do not guarantee the result. I can pass one set of Id's to this method and get a response. Then pass a completely different set of Ids and it will return back the same response from the first call. Given that, what is the value of storing the result in a static var here? The calling method should be the only place responsible for caching the result.

Reply:
Yep - I think I did this when trying to get to the bottom of the issue with the merge recursion that led to the introduction of that "timesRun" flag.  I'll take these vars back out.